engine/common/provider/engine.go

```diff
@@ -210,7 +236,7 @@ func (e *Engine) onEntityRequest(originID flow.Identifier, req *messages.EntityR
 
 	// send back the response
 	res := &messages.EntityResponse{
-		Nonce:     req.Nonce,
+		Nonce:     rand.Uint64(),
```

Comment:
why reset it here? I thought the purpose of the nonce was to correlate request/responses on the requester side

Reply:
https://github.com/onflow/flow-go/pull/3404/commits/1793639ab5041665d21f8ad3b8c1a9ff538bfce3